.github/actions/run/action.yml

```diff
@@ -0,0 +1,76 @@
+name: "Run (with or without container)"
+description: "Run in a container, or not."
+inputs:
+  container:
+    description: The container address
+  script:
+    description: The script to run
+  test-outcomes-dir:
+    description: The test outcomes directory
+  test-device-addr:
+    description: The test device address to use
+
+runs:
+  using: "composite"
+  steps:
+    - name: Run
+      if: ${{ inputs.container == '' }}
+      shell: bash
+      env:
+        SCRIPT: ${{ inputs.script }}
+      run: ferrocene/ci/run.sh
+
+    - name: Pull container
+      if: ${{ inputs.container != '' }}
+      shell: bash
+      run: docker pull --quiet ${{ inputs.container }}
+
+    - name: Run in container
+      if: ${{ inputs.container != '' }}
+      # Workaround for https://github.com/actions/runner/issues/241
+      shell: script --return --quiet --log-out /dev/null --command "bash -e {0}"
+      env:
+        TEST_DEVICE_ADDR: ${{ inputs.test-device-addr }}
+        SCRIPT: |
+          ./ferrocene/ci/scripts/setup-uv.sh
+          ${{ inputs.script }}
+        FERROCENE_TEST_OUTCOMES_DIR: ${{ inputs.test-outcomes-dir }}
```

Comment:
I kinda don't like that we download the test outcomes outside of this composite action, and then pass as a parameter the path to the test outcomes.

I'd prefer if we passed `test-outcomes: true` to a job, and the composite action would download the outcomes and pass the correct path.

Reply:
So in general Actions don't accept boolean style arguments well. If you pass `thing: true` that's all well and fine but if you don't pass anything it passes `thing: ''` which, well, is not false when I checked. :(

Reply:
I resolved this by making `configure:` a propr arg and removing the `test-outcomes-dir` arg entirely.

Reply:
`RUST_CONFIGURE_ARGS` has been removed in the last iteration of `ferrocene/ci/configure.sh` :sweat_smile: 

My main point of this review comment was to download the test outcomes inside of the `run` action, so that the code for it isn't duplicated across multiple workflows. For that to work, we'd still need to have an equivalent of `test-outcomes: true`. I guess we can then check it with `inputs.test-outcomes != ''`.